pyiron_workflow/type_hinting.py

```diff
@@ -28,7 +28,12 @@ def valid_value(value, type_hint) -> bool:
 
 
 def type_hint_to_tuple(type_hint) -> tuple:
-    if isinstance(type_hint, types.UnionType | typing._UnionGenericAlias):
+    if isinstance(
+        type_hint,
+        types.UnionType | typing._UnionGenericAlias,  # type: ignore
```

Comment:
Do we need to rely on some internal type?

Reply:
No, good catch. `typing._UnionGenericAlias` was necessary for python<3.10 (to handle the `|` in `a | b` IIRC), but we dropped 3.9 support so we can just get rid of this.

I patched it here: https://github.com/pyiron/pyiron_workflow/pull/535